File: lowestcommonancestor/LCAPython/LCA.py
class Node:
    def __init__(self, key):
        self.key = key
        self.left = None
        self.right = None

def findLCA(root, n1, n2):

    if root is None:
        return None

    if root.key == n1 or root.key == n2:
        return root

    left_lca = findLCA(root.left, n1, n2)
    right_lca = findLCA(root.right, n1, n2)

    if left_lca and right_lca:
        return root

    return left_lca if left_lca is not None else right_lca

File: lowestcommonancestor/LCA/src/LCA.java
public class LCA
{
    Node root;

    Node findLCA(int n1, int n2)
    {
        return findLCA(root,n1,n2);
    }
    Node findLCA(Node node,int n1,int n2)
    {
        if(node == null)
        {
            return null;
        }
        if (node.data == n1 || node.data == n2)
        {
            return node;
        }

        Node left_lca = findLCA(node.left, n1, n2); 	
	    Node right_lca = findLCA(node.right, n1, n2);

        if (left_lca!=null && right_lca!=null) 
        {
            return node; 
        }       

        return (left_lca != null) ? left_lca : right_lca;	
	            
    }

}
